### ISSUE TITLE
Support the ability to dynamically configure new apps

### ISSUE THREAD

**Issue by kevin-ashton** [wakatime/macos-wakatime#273]
### BundleId of the app

anything.example.com

### Example window titles

We want to use Wakatime more at my company but the apps supported are too limited. Seems strange there isn't a way to just dynamically add support for an app.  Does the app id really have to be hard coded into the build?  I mean if I downloaded Chrome and the only supported URLs had to be hard coded into the browser that would be a bit weird. 

Anyway this request is to just allow users to dynamically add apps they want, even if they have to manually enter the app id.

**Comment by alanhamlett** [comment]
Since we detect the file name from the app's a11y dom, we need to add custom code to support parsing the currently focused file and project for each new app.

We can let you track any app, but it will only use the window title for the file name and won't be able to detect a project.

**Comment by kevin-ashton** [comment]
I agree that extra data is nice, but the title is fine often times. For example for us it is often good enough to just know roughly where we are spending our time. 

If you add this feature perhaps you can make make it clear there are limits to what data can be collected.
